### PR TITLE
[SAGE-410] Toolbar updates

### PR DESCRIPTION
**Pull request by monkeypox8**
## Description
Style modifications to align toolbar content with [Figma mockups](https://www.figma.com/file/sMbtLUHSt2vfKgKjyQ3052/Sage?node-id=3661%3A24050).

## Screenshots

| Sage 3 |Before (Sage Next)  |  After (Sage Next)  |
|---|-----|--------|
|before-sage3|before-sage-next|after-sage-next|

## Testing in `sage-lib`
1. View the [Toolbar](http://localhost:4000/pages/component/toolbar?sage_theme=sage_theme_next)
2. Confirm that Button, Search, Input, and Dropdown styles are contained and styled as expected

## Testing in `kajabi-products`

1. (**LOW**) Toolbar: style modifications. New component, no expected impact in KP. Some related style updates in other components that should be limited to Toolbar placement.

### CODE REVIEW

**Review by monicawheeler — APPROVED** [review]
🙌

File: docs/app/views/examples/components/themes/next/toolbar/_preview.html.erb [hunk 1]
```diff
@@ -1,63 +1,56 @@
...
+  }
+] %>
 
-<h3>Toolbar with supported items</h3>
 <%= md("
   Toolbars currently support secondary buttons, checkboxes, dropdowns, search, and plain text.
   Each of these can also be placed inside a nested `SageToolbarGroup` to merge items with borders.
 ", use_sage_type: true) %>
+
+<h3 class="t-sage-heading-6">Basic toolbar</h3>
```
Comment (pixelflips) [inline docs/app/views/examples/components/themes/next/toolbar/_preview.html.erb:48]: Could/should we use `SageClassnames::TYPE::HEADING_6` for the headings?

File: packages/sage-assets/lib/stylesheets/themes/next/components/_button.scss [hunk 2]
```diff
@@ -16,6 +16,7 @@ $-btn-icon-only-standard-size: rem(40px);
 $-btn-icon-only-subtle-size: rem(16px);
 $-btn-icon-only-hover-size: rem(24px);
 $-btn-icon-only-focus-size: rem(32px);
+$-btn-toolbar-group-height: rem(40px);
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/themes/next/components/_button.scss:19]: There's some odd behavior at small viewports when a button with an icon is added to the Toolbar. The icon's height extends 2-3px past the button's bounds, causing it to appear larger than the Toolbar.

File: docs/app/views/examples/components/themes/next/toolbar/_preview.html.erb [hunk 3]
```diff
@@ -1,63 +1,56 @@
...
+    icon: "add",
+    style: "primary",
+    attributes: { "href": "#" },
+  }, {
+    value: "Share Element",
+    icon: "url",
+    modifiers: ["border-before"],
+    attributes: { "href": "#" },
+  }, {
+    value: "Take A Dangerous Action",
```
Comment (pixelflips) [inline docs/app/views/examples/components/themes/next/toolbar/_preview.html.erb:28]: This one seems to wrap quite a bit in the examples. Not a show stopper, but wondering if we should check w/ design about the reduction of the new `.sage-dropdown__menu` min-width? Just leaving a note here and this should not stop this PR.

File: packages/sage-assets/lib/stylesheets/themes/next/components/_button.scss [hunk 4]
```diff
@@ -285,32 +292,46 @@ $-btn-loading-min-height: rem(36px);
...
+  .sage-panel-controls__toolbar-btn-group > &:last-child,
+  .sage-toolbar__group > &:last-child,
   .sage-toolbar__group > .sage-dropdown:last-child .sage-dropdown__trigger > & {
     border-top-right-radius: sage-border(radius);
     border-bottom-right-radius: sage-border(radius);
   }
 
+  .sage-toolbar__group > .sage-dropdown + &,
+  .sage-toolbar__group > & + & {
+    margin-left: rem(-1px); /* offset border overlap of multiple buttons */
```
Comment (pixelflips) [inline packages/sage-assets/lib/stylesheets/themes/next/components/_button.scss:328]: ❤️

File: packages/sage-assets/lib/stylesheets/themes/next/components/_search.scss [hunk 5]
```diff
@@ -48,7 +46,8 @@ $-search-input-height: rem(40px);
     z-index: sage-z-index(default);
     border-radius: sage-border(radius-medium);
     box-shadow: map-get($sage-toolbar-button-borders, default);
-    border: 0;
+    transition: map-get($sage-transitions, input);
+    transition-property: border, box-shadow;
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/themes/next/components/_search.scss:50]: Temporary transition styles to match Form Input behavior. The Rails markup structure of Search is something of an outlier from Input due to its embedded `form`, and the stylesheet is in need of an overhaul.

File: packages/sage-assets/lib/stylesheets/themes/next/components/_search.scss [hunk 6]
```diff
@@ -129,8 +134,10 @@ $-search-input-height: rem(40px);
   }
 
   .sage-panel-controls__toolbar &,
+  .sage-toolbar__group &,
   .sage-search--contained & {
-    height: $-search-input-height;
+    padding-top: sage-spacing(xs);
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/themes/next/components/_search.scss:139]: Replaced the defined `height` with padding here to allow height constraints via the container. Open to feedback on whether we see this being necessary.

**Review by pixelflips — CHANGES_REQUESTED** [review]
Overall this looks and works great. 

Left a couple of comments and noticed one minor issue that on smaller screens the buttons w/ icons and inputs don't quite align. I think this is related to the padding applied to the icons.

Screen Shot 2022-04-22 at 11 25 59 AM

File: packages/sage-assets/lib/stylesheets/themes/next/components/_button.scss (hunk 2 above)
Comment (pixelflips) [reply packages/sage-assets/lib/stylesheets/themes/next/components/_button.scss:19]: @monkeypox8 noticed this comment after submitting my review. I'm happy to give this one a good-to-go 👍🏼  after seeing this comment.

**Review by pixelflips — APPROVED** [review]
LGTM!

File: docs/app/views/examples/components/themes/next/toolbar/_preview.html.erb (hunk 3 above)
Comment (monkeypox8) [reply docs/app/views/examples/components/themes/next/toolbar/_preview.html.erb:28]: Good callout. I'm not certain what the reasoning was behind the decreased width on Dropdowns but worth a follow-up.

**Pull request merged**